Plugins/Power Manager/Metrics & Monitoring/Python/get_power_manager_top_energy_consumers.py

```diff
@@ -0,0 +1,206 @@
+                if 'error' in top_energy_consumer_json_data:
+                    error_content = top_energy_consumer_json_data['error']
+                    if '@Message.ExtendedInfo' not in error_content:
+                        print("Unable to retrieve Power Manager - top energy consuming %s in %s" % (entity,ip_address))
+                    else:
+                        extended_error_content = error_content['@Message.ExtendedInfo']
+                        print("Unable to retrieve Power Manager - top energy consuming %s in %s. See below ExtendedInfo for more information" % (entity,ip_address))
+                        print(extended_error_content[0]['Message'])
+                else:
+                    print("Unable to retrieve Power Manager - top energy consuming %s in %s" % (entity,ip_address))
+            else:
+            
+                top_energy_consumer_content = json.loads(top_energy_consumer_response.content)
+                
+                if 'Value' not in top_energy_consumer_content:
+                    print("No Power Manager - top energy consuming %s in %s" % (entity,ip_address))
+                else:
+                    # For every elements in the Power Manager Metrics Alerts response, store the details in the table
+                    for top_energy_consumer_elem in top_energy_consumer_content["Value"]:
+                        
+                        top_energy_consumer_data = [top_energy_consumer_elem["Id"], top_energy_consumer_elem["Name"], top_energy_consumer_elem["Value"]]
+                        top_energy_consumer_output_column_data.append(top_energy_consumer_data)
+                        
+                    table = columnar(top_energy_consumer_output_column_data, output_column_headers, no_borders=True)
+                    print("\n   =====================================================")
+                    print("      Power Manager - Top energy consuming %s " % (entity))
+                    print("   =====================================================")
+                    print(table)
+    except:
+        print ("Unexpected error:", sys.exc_info()[0])
+
+
+if __name__ == '__main__':
+    PARSER = argparse.ArgumentParser(description=__doc__,
+                                     formatter_class=RawTextHelpFormatter)
+    PARSER.add_argument("--ip", "-i", required=True, help="OpenManage Enterprise  IP")
+    PARSER.add_argument("--username", "-u", required=False, help="Username for OpenManage Enterprise ", default="admin")
+    PARSER.add_argument("--password", "-p", required=True, help="Password for OpenManage Enterprise ")
+    ARGS = PARSER.parse_args()
+    
+    get_inputs = 'Y'
+    
+    #Until user says NO to get top energy consumers
+    while get_inputs in ('Y','y',''):
+        
+        #Get the top energy consumers choice
+        entity_inputs = input("Enter the choice of Top Energy Consumer - 1 - Server / 2 - Chassis / 3 - Group : ")
```

Comment:
Shouldn't this be a parameter for the script - why would an user be prompted again for this data?